### PR TITLE
Fix error with find hander protocol calls when creating new lambda

### PR DESCRIPTION
**Pull request by sdubov**
* Optimized logic to get handlers inside a solution
* Update call for find handler logic to sync call to make sure response is updated on a correct thread

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Description
Fix issue with creating a new AWS Lambda and validating a handler inside a configuration panel.

## Motivation and Context
The main purpose for this commit is to fix the issue with validating a handler in a backend PSI while creating a new AWS Lambda. This issue prevents from creating a Lambda because of panel validation error.

## Related Issue(s)
None.

## Testing
Checked new AWS Lambda creation logic.

## Screenshots (if appropriate)
None.

## Checklist
- [x] I have read the **README** document
- [x] I have read the **CONTRIBUTING** document
- [x] Local run of `gradlew check` succeeds
- [x] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed
- [ ] A short description of the change has been added to the **CHANGELOG**

## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jetbrains-rider/ReSharper.AWS/src/AWS.Daemon/Lambda/LambdaFinder.cs [hunk 1]
```diff
@@ -0,0 +1,336 @@
...
+using JetBrains.ProjectModel.ProjectsHost.Dependencies;
+using JetBrains.ReSharper.Psi;
+using JetBrains.ReSharper.Psi.CSharp;
+using JetBrains.ReSharper.Psi.Resolve;
+using JetBrains.ReSharper.Psi.Util;
+using JetBrains.UI.ThemedIcons;
+using JetBrains.Util;
+using JetBrains.Util.Logging;
+
+namespace ReSharper.AWS.Lambda
```
Comment (abrooksv) [inline jetbrains-rider/ReSharper.AWS/src/AWS.Daemon/Lambda/LambdaFinder.cs]: What tests covers this logic?

**Comment by abrooksv** [comment]
Production code looks good, just question around what tests hit that logic

File: jetbrains-rider/ReSharper.AWS/src/AWS.Daemon/Lambda/LambdaFinder.cs (hunk 1 above)
Comment (sdubov) [reply jetbrains-rider/ReSharper.AWS/src/AWS.Daemon/Lambda/LambdaFinder.cs]: This is a pease of refactoring for defining lambda handlers. This logic is used LambdaGutterMarkHighlighting and is covered by LambdaGutterMarkHighlightingTest. I left the duplicate code in the original commit. Removed it + removed the unused method to  collect all handlers from backend logic to not mix it for now. Sorry for confusion. Pushed an updated.

Comment (abrooksv) [reply jetbrains-rider/ReSharper.AWS/src/AWS.Daemon/Lambda/LambdaFinder.cs]: Okay, sounds good.

**Pull request merged**